Tests/L0/SonarQubePreBuild/_suite.ts

```diff
@@ -0,0 +1,30 @@
+/// <reference path="../../../definitions/mocha.d.ts"/>
+/// <reference path="../../../definitions/node.d.ts"/>
+/// <reference path="../../../definitions/Q.d.ts"/>
+
```

Comment:
You'll notice that the test "driver" is actually nodejs. I believe the testing framework used is Jasmine http://jasmine.github.io/1.3/introduction.html - this makes a lot of sense if you consider that tasks are moving to a nodejs-only implementation. 

Some tasks - like msbuild and the SonarQube ones - will retain their PowerShell-only implementation so all this javascript is just hosting powershell tests. See https://github.com/Microsoft/vso-agent-tasks/tree/master/Tasks/Grunt for an example of testing a nodejs task (it's quite a bit cleaner).